readthedocs/search/api/v3/views.py

```diff
@@ -0,0 +1,134 @@
+from functools import cached_property
+
+import structlog
+from django.utils import timezone
+from django.utils.translation import gettext as _
+from rest_framework.exceptions import ValidationError
+from rest_framework.generics import GenericAPIView
+
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.search import tasks
+from readthedocs.search.api.pagination import SearchPagination
+from readthedocs.search.api.v3.backend import Backend
+from readthedocs.search.api.v3.serializers import PageSearchSerializer
+from readthedocs.search.api.v3.utils import should_use_advanced_query
+
+log = structlog.get_logger(__name__)
+
+
+class SearchAPI(GenericAPIView):
+
+    """
+    Server side search API V3.
+
+    Required query parameters:
+
+    - **q**: [Search term](https://docs.readthedocs.io/page/server-side-search/syntax.html).
+
+    Check our [docs](https://docs.readthedocs.io/page/server-side-search/api.html) for more information.
+    """  # noqa
+
+    http_method_names = ["get"]
+    pagination_class = SearchPagination
+    serializer_class = PageSearchSerializer
+    search_backend_class = Backend
+
+    def get_view_name(self):
+        return "Search API V3"
+
+    def _validate_query_params(self):
+        query = self.request.GET.get("q")
+        errors = {}
+        if not query:
+            errors["q"] = [_("This query parameter is required")]
+        if errors:
+            raise ValidationError(errors)
+
+    @cached_property
+    def _backend(self):
+        backend = self.search_backend_class(
+            request=self.request,
+            query=self.request.GET["q"],
+        )
+        return backend
+
+    def _get_search_query(self):
+        return self._backend.parser.query
+
+    def _get_projects_to_search(self):
+        return self._backend.projects
```

Comment:
I'm not sure if wrapping this makes it more readable. Probably easiest to just include them directly, unless we expect this logic to change?

Reply:
It felt more explicit since we are using it in several places inside this class